examples/azure/terraform-azure-availabilityset-example/main.tf

```diff
@@ -0,0 +1,104 @@
+provider "azurerm" {
+  version = "=2.20.0"
+  features {}
+}
+
+# ---------------------------------------------------------------------------------------------------------------------
+# PIN TERRAFORM VERSION TO >= 0.12
+# The examples have been upgraded to 0.12 syntax
+# ---------------------------------------------------------------------------------------------------------------------
+
+terraform {
+  required_version = ">= 0.12"
+}
+
+# ---------------------------------------------------------------------------------------------------------------------
+# DEPLOY A RESOURCE GROUP
+# See test/terraform_azure_example_test.go for how to write automated tests for this code.
+# ---------------------------------------------------------------------------------------------------------------------
+
+resource "azurerm_resource_group" "main" {
+  name     = "${var.prefix}-resources"
+  location = var.location
+}
+
+# ---------------------------------------------------------------------------------------------------------------------
+# DEPLOY AVAILABILITY SET
+# ---------------------------------------------------------------------------------------------------------------------
+
+resource "azurerm_availability_set" "main" {
+  name                        = "${var.prefix}-example"
+  location                    = azurerm_resource_group.main.location
+  resource_group_name         = azurerm_resource_group.main.name
+  platform_fault_domain_count = var.avs_fault_domain_count
+  managed                     = true
+}
+
+# ---------------------------------------------------------------------------------------------------------------------
+# DEPLOY VIRTUAL NETWORK RESOURCES
```

Comment:
Is this necessary? Or could a default network be used to reduce the complexity of this example? For all our AWS examples, we always use the default VPC.

Reply:
In Azure there is no default network, hence the need to either provision one in the module or reference an already deployed virtual network from another deployment.  We wanted to make the example self contained and hence the vnet is in the example.